api/editUser.go

```diff
@@ -0,0 +1,46 @@
+package editUser
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/mhghw/fara-message/db"
+)
+
+func editUser(c *gin.Context) {
+	var username string
+
+	err := c.BindJSON(&username)
+	if err != nil {
+		fmt.Errorf("error binding JSON:%w", err)
+		c.Status(400)
+		return
+	}
```

Comment:
You can't bind json with string, this JSON body has a key and a value and you should create a good struct for that.